### PR TITLE
Task av/j50 attach photo for podcast functionality

### PR DESCRIPTION
**Pull request by artem-totality**

### CODE REVIEW

**Review by Sayonarko — DISMISSED** [review]

File: frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if (file) {
+      onSubmit(file);
+    }
+  };
+
+  return (
+    <form>
+      <fieldset disabled={isFormDisabled} className={styles.fieldset}>
+        <img
+          src={cover ? cover.url : logoCut}
```
Comment (yevhenprybytok) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx]:
```suggestion
          src={cover?.url ?? logoCut}
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { useAppSelector, useRef } from 'hooks/hooks';
```
Comment (yevhenprybytok) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx]: Is this component can only be used in the podcast page?

File: frontend/src/services/image-api/image-api.service.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  #datauri: Datauri;
+  #apiPrefix: string;
+
+  constructor({ http, datauri, apiPrefix }: Constructor) {
+    this.#http = http;
+    this.#datauri = datauri;
+    this.#apiPrefix = apiPrefix;
+  }
+
+  public async uploadImage(file: File, userId = 1): Promise<Image> {
```
Comment (yevhenprybytok) [inline frontend/src/services/image-api/image-api.service.ts]: What's this userId for?

**Review by yevhenprybytok — CHANGES_REQUESTED** [review]

File: backend/src/api/api.ts [hunk 4]
```diff
@@ -7,13 +7,15 @@ import {
...
   record as recordService,
+  fileStorage as fileStorageService,
 } from '~/services/services';
 import { initAuthApi } from './auth/auth.api';
 import { initUsersApi } from './users/users.api';
 import { initPodcastsApi } from './podcasts/podcasts.api';
 import { initEpisodesApi } from './episodes/episodes.api';
 import { initCommentsApi } from './comments/comments.api';
 import { initRecordsApi } from './records/records.api';
+import { initFileApi } from './image/image.api';
```
Comment (ghost) [inline backend/src/api/api.ts]:
```suggestion
```

File: backend/src/api/image/image.api.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import { Router } from 'express';
```
Comment (ghost) [inline backend/src/api/image/image.api.ts]: we don't want this api

File: frontend/src/store/configurate-podcast/reducer.ts [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
 import { DataStatus } from 'common/enums/enums';
-import { Podcast } from 'common/types/types';
-import { create } from './actions';
+import { Podcast, Image } from 'common/types/types';
+import { create, createCover } from './actions';
 
 type State = {
   dataStatus: DataStatus;
   podcast: Podcast | null;
+  image: Image | null;
```
Comment (ghost) [inline frontend/src/store/configurate-podcast/reducer.ts]:
```suggestion
```

File: frontend/src/store/configurate-podcast/actions.ts [hunk 7]
```diff
@@ -14,4 +15,12 @@ const create = createAsyncThunk<Podcast, PodcastCreatePayload, AsyncThunkConfig>
...
 });
 
-export { create };
+const createCover = createAsyncThunk<Image, File, AsyncThunkConfig>
+(ActionType.CREATE_COVER, async (imageFile, { extra }) => {
+  const { imageApi } = extra;
+  const cover = await imageApi.uploadImage(imageFile);
+
+  return cover;
+});
```
Comment (ghost) [inline frontend/src/store/configurate-podcast/actions.ts]:
```suggestion
```

File: frontend/src/services/image-api/image-api.service.ts [hunk 8]
```diff
@@ -0,0 +1,58 @@
+import {
```
Comment (ghost) [inline frontend/src/services/image-api/image-api.service.ts]: we don't want this service

**Review by ghost — CHANGES_REQUESTED** [review]

File: shared/src/validation-schemas/image/image.validation-schema.ts [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import * as Joi from 'joi';
```
Comment (ghost) [inline shared/src/validation-schemas/image/image.validation-schema.ts]: remove

File: shared/src/common/types/image/image-upload.type.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { ImageUploadKey } from '~/common/enums/enums';
+
+type ImageUploadPayload = {
```
Comment (ghost) [inline shared/src/common/types/image/image-upload.type.ts]: remove it

File: shared/src/common/enums/validation/image-validation-message.enum.ts [hunk 11]
```diff
@@ -0,0 +1,7 @@
+const ImageValidationMessage = {
```
Comment (ghost) [inline shared/src/common/enums/validation/image-validation-message.enum.ts]: remove it

File: shared/src/common/enums/image/image-upload-key.enum.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+enum ImageUploadKey {
```
Comment (ghost) [inline shared/src/common/enums/image/image-upload-key.enum.ts]: remove it

File: frontend/src/store/configurate-podcast/actions.ts [hunk 13]
```diff
@@ -6,6 +6,7 @@ import {
 } from 'common/types/types';
 import { ActionType } from './common';
 
+// prettier-ignore
```
Comment (ghost) [inline frontend/src/store/configurate-podcast/actions.ts]:
```suggestion
```

File: frontend/src/services/datauri/datauri.service.ts [hunk 14]
```diff
@@ -0,0 +1,11 @@
+class Datauri {
```
Comment (ghost) [inline frontend/src/services/datauri/datauri.service.ts]: just create a helper `getDataUrl` and move it to the `frontend/src/helpers/file`

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/data/models/podcast/podcast.model.ts [hunk 15]
```diff
@@ -13,6 +15,19 @@ class Podcast extends Abstract {
   static get tableName(): string {
     return TableName.PODCASTS;
   }
+
+  static get relationMappings(): any {
```
Comment (ghost) [inline backend/src/data/models/podcast/podcast.model.ts]:
```suggestion
  static relationMappings = {
```

File: frontend/src/helpers/file/get-data-url/get-data-url.ts [hunk 16]
```diff
@@ -0,0 +1,9 @@
+const getDataUrl = async (file: File): Promise<string> => {
```
Comment (ghost) [inline frontend/src/helpers/file/get-data-url/get-data-url.ts]:
```suggestion
const getDataUrl = (file: File): Promise<string> => {
```

File: frontend/src/helpers/file/get-data-url/get-data-url.ts [hunk 17]
```diff
@@ -0,0 +1,9 @@
...
+  return new Promise((resolve) => {
```
Comment (ghost) [inline frontend/src/helpers/file/get-data-url/get-data-url.ts]:
```suggestion
  return new Promise((resolve, reject) => {
```

File: frontend/src/helpers/file/get-data-url/get-data-url.ts [hunk 18]
```diff
@@ -0,0 +1,9 @@
...
+    const fileReader = new FileReader();
+    fileReader.onload = (): void => resolve(<string>fileReader.result);
+    fileReader.readAsDataURL(file);
```
Comment (ghost) [inline frontend/src/helpers/file/get-data-url/get-data-url.ts]:
```suggestion
    fileReader.addEventListener('load', () => {
      resolve(<string>fileReader.result)
    }, {
      once: true
    });
    
    fileReader.addEventListener('error', () => {
      reject(fileReader.error)
    }, {
      once: true
    });
    
    fileReader.readAsDataURL(file);
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+
+const ConfiguratePodcastImage: React.FC<Props> = ({ onSubmit, imageSrc }) => {
+  const { createPodcastStatus } = useAppSelector(({ configuratePodcast }) => ({
+    createPodcastStatus: configuratePodcast.dataStatus,
+  }));
+
+  const isFormDisabled = createPodcastStatus === DataStatus.PENDING;
+
+  const handleCoverUpload = (evt: React.FormEvent<HTMLInputElement>): void => {
+    const file = evt.currentTarget?.files?.[0];
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx]:
```suggestion
    const [file] = evt.currentTarget?.files ?? [];
```

File: frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+        <img
+          src={imageSrc ? imageSrc : logoCut}
+          width={195}
+          height={195}
+          loading="lazy"
+          alt="image"
+        />
+        <input
+          type="file"
+          accept=".jpg, .png, .svg"
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx]: create enum for it and also use it in the validation

File: frontend/src/common/types/types.ts [hunk 21]
```diff
@@ -3,5 +3,6 @@ export type { User, UserCreatePayload, UserSignInPayload } from './user/user';
 export type { HttpOptions } from './http/http';
 export type { ValidationSchema } from './validation-schema/validation-schema';
 export type { Podcast, PodcastCreatePayload } from './podcast/podcast';
+export type { Image } from './image/image';
```
Comment (ghost) [inline frontend/src/common/types/types.ts]:
```suggestion
```

File: frontend/src/common/types/image/image.ts [hunk 22]
```diff
@@ -0,0 +1 @@
+export type { Image } from 'jabber-shared/common/types/types';
```
Comment (ghost) [inline frontend/src/common/types/image/image.ts]: remove file

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 23]
```diff
@@ -1,10 +1,17 @@
-import { Podcast as TPodcast, PodcastCreatePayload } from '~/common/types/types';
+import { Podcast as TPodcast } from '~/common/types/types';
 import { PodcastModel as PodcastM } from '~/data/models/models';
 
 type Constructor = {
   PodcastModel: typeof PodcastM;
 };
 
+type PodcastRecordPayload = {
```
Comment (ghost) [inline backend/src/data/repositories/podcast/podcast.repository.ts]:
```suggestion
type PodcastCreateDTOPayload = {
```
move it to the common folder on the backend

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/data/repositories/podcast/podcast.repository.ts [hunk 24]
```diff
@@ -1,10 +1,17 @@
-import { Podcast as TPodcast, PodcastCreatePayload } from '~/common/types/types';
+import { Podcast as TPodcast } from '~/common/types/types';
 import { PodcastModel as PodcastM } from '~/data/models/models';
 
 type Constructor = {
   PodcastModel: typeof PodcastM;
 };
 
+type PodcastCreateDTOPayload = {
```
Comment (ghost) [inline backend/src/data/repositories/podcast/podcast.repository.ts]: move it to `backend/src/common/types`

File: frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx [hunk 25]
```diff
@@ -0,0 +1,44 @@
...
+
+  const handleCoverUpload = (evt: React.FormEvent<HTMLInputElement>): void => {
+    const [file] = evt.currentTarget?.files ?? [];
+
+    if (file) {
+      onSubmit(file);
+    }
+  };
+
+  const fileMask = Object.values(ImageFileExtension).join(', ');
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/components/configurate-podcast-image/configurate-podcast-image.tsx]: move it outside of the component

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/configurate-podcast/configurate-podcast.tsx [hunk 26]
```diff
@@ -12,15 +16,33 @@ const ConfiguratePodcast: React.FC = () => {
   const isEdit = Boolean(id);
 
   const handleCreatePodcast = (payload: PodcastCreatePayload): void => {
-    dispatch(configuratePodcastActions.create(payload));
+    dispatch(
+      configuratePodcastActions.create({ ...payload, imgDataUrl: imageSrc }),
+    );
+  };
+
+  const [imageSrc, setImageSrc] = useState('');
```
Comment (ghost) [inline frontend/src/components/configurate-podcast/configurate-podcast.tsx]: move stats before `handleCreatePodcast `

**Review by ghost — CHANGES_REQUESTED** [review]

**Pull request merged**